streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/StandbyTaskAssignmentUtils.java

```diff
@@ -49,7 +53,11 @@ static int pollClientAndMaybeAssignAndUpdateRemainingStandbyTasks(final Map<UUID
             tasksToRemainingStandbys.put(activeTaskId, numRemainingStandbys);
         }
 
-        return numRemainingStandbys;
+        log.warn("Unable to assign {} of {} standby tasks for task [{}]. " +
```

Comment:
Refactored based on previous PR comment
https://github.com/apache/kafka/pull/10851#discussion_r817629787

Since both standby task assignors have the same logging, moved the logging to this utility method.

Reply:
This doesn't look right. We log the warning to user even when `numRemainingStandbys==0`?

Reply:
Stupid mistake. Fixed. Also added tests.